src/types/package.rs

```diff
@@ -359,6 +360,19 @@ pub struct Package {
     pub developer_responsiveness: Option<DeveloperResponsiveness>,
     pub issue_impacts: IssueImpacts,
     pub complete: bool,
+    pub release_data: Option<PackageReleaseData>,
+    pub repo_url: Option<String>,
+    #[serde(rename = "maintainers_recently_changed")]
+    pub maintainers_recently_changed: Option<bool>,
+    #[serde(rename = "is_abandonware")]
+    pub is_abandonware: Option<bool>,
```

Comment:
Could we maybe make these aliases and encourage the API to change them to camel case? Not really a big fan of mixing up our capitalization at random.

Reply:
We could, but then the CLI will serialize them differently so the output will be different between curl and `--json`.

Reply:
I guess it depends a bit on our goal at that point. Do we actually *want* these to be snake case or is that a bug that will hopefully get fixed in the future?

If it's a bug that will get fixed in the future, I'd suggest going with an alias. If we *want* these to be snake case... I don't understand why. So I'd be fine with the rename if we don't plan to change them, but considering that hopefully that's a very realistic option, I think it's much better for forward-compatibility to use aliases here?

Reply:
Also I don't think we make any guarantee that our CLI's json output shares any similarity to our API's output. So I'm not sure that's necessary.

Reply:
It doesn't look like this is possible. If I put `#[serde(alias = "maintainers_recently_changed")]` and then try to deserialize JSON that uses the correct capitalization I get an error "duplicate field `maintainersRecentlyChanged`". We need to use `rename` for now until after the correct fields are returned from the API result.

Reply:
For transition, you may need to supply both `rename` and `alias` (so that deserialization works no matter which name serialization uses)

```suggestion
    #[serde(rename = "maintainers_recently_changed")]
    #[serde(alias = "maintainersRecentlyChanged")]
    pub maintainers_recently_changed: Option<bool>,
    #[serde(rename = "is_abandonware")]
    #[serde(alias = "isAbandonware")]
    pub is_abandonware: Option<bool>,
```

Reply:
I think we're trying to solve this compatibility issue from different directions. I have a PR out so a future version of the API will mark the snake case version as deprecated in favor of a new camel case version. Both capitalizations will be returned at the same time. This maintains compatibility with existing clients and clients that don't know we plan to change the capitalization later.

```Rust
:dep serde = { version = "1", features = ["derive"] }
:dep serde_json = "1"
```


```Rust
use serde::Deserialize;
use serde_json::json;
```


```Rust
fn check_compatibility<T>()
where
    T: for <'a> Deserialize<'a>,
{
    for (name, value) in [
        (
            "old",
            json!({
                "is_abandonware": true,
            }),
        ),
        (
            "transition",
            json!({
                "is_abandonware": true,
                "isAbandonware": true,
            }),
        ),
        (
            "future",
            json!({
                "isAbandonware": true,
            }),
        ),
    ] {
        match serde_json::from_value::<T>(value) {
            Ok(_) => println!("✅ {}: OK", name),
            Err(err) => println!("❌ {}: {}", name, err),
        }
    }
}
```


```Rust
#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
struct Current {
    #[serde(rename = "is_abandonware")]
    is_abandonware: bool,
}
check_compatibility::<Current>()
```

    ✅ old: OK
    ✅ transition: OK
    ❌ future: missing field `is_abandonware`





    ()




```Rust
#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
struct Alias {
    #[serde(alias = "is_abandonware")]
    is_abandonware: bool,
}
check_compatibility::<Alias>()
```

    ✅ old: OK
    ❌ transition: duplicate field `isAbandonware`
    ✅ future: OK





    ()




```Rust
#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
struct RenameAndAlias {
    #[serde(rename = "is_abandonware")]
    #[serde(alias = "isAbandonware")]
    is_abandonware: bool,
}
check_compatibility::<RenameAndAlias>()
```

    ✅ old: OK
    ❌ transition: duplicate field `is_abandonware`
    ✅ future: OK





    ()

Reply:
And then, obviously,


```Rust
#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
struct Final {
    is_abandonware: bool,
}
check_compatibility::<Final>()
```

    ❌ old: missing field `isAbandonware`
    ✅ transition: OK
    ✅ future: OK





    ()

Reply:
> Both capitalizations will be returned at the same time.

Yep. This is what I missed. So once that PR is merged and deployed, we can remove the rename here.

That works for me, but we should probably have an issue or something to track it so that we don't forget about it.

Reply:
Created #52 to collect stuff like this. If there's more we can add to that ticket.